beacon-chain/p2p/broadcaster.go

```diff
@@ -57,19 +60,29 @@ func (s *Service) BroadcastAttestation(ctx context.Context, subnet uint64, att *
 		return err
 	}
 	// Ensure we have peers with this subnet.
-	if !s.hasPeerWithSubnet(subnet) {
-		for {
-			if err := ctx.Err(); err != nil {
-				return err
-			}
-			ok, err := s.FindPeersWithSubnet(ctx, subnet)
-			if err != nil {
-				return err
-			}
-			if ok {
-				savedAttestationBroadcasts.Inc()
-				break
+	s.subnetLocker(subnet).RLock()
+	hasPeer := s.hasPeerWithSubnet(subnet)
+	s.subnetLocker(subnet).RUnlock()
+	if !hasPeer {
+		if err := func () error {
+			s.subnetLocker(subnet).Lock()
+			defer s.subnetLocker(subnet).Unlock()
+			for i := 0; i < maxSubnetDiscoveryAttempts; i++ {
+				if err := ctx.Err(); err != nil {
+					return err
+				}
+				ok, err := s.FindPeersWithSubnet(ctx, subnet)
+				if err != nil {
+					return err
+				}
+				if ok {
+					savedAttestationBroadcasts.Inc()
+					break
+				}
 			}
+			return nil
```

Comment:
Shouldnt we return an error at least ? If after 3 tries, we still cannot find a peer to broadcast the attestation to.